### DISCUSSION TITLE
change nickname on a modal

### DISCUSSION THREAD

**Discussion by k3nnyburger** [Rapptz/discord.py#9085] [Q&A] [answered]
I wanted to change my modal a nickname but unfortunately do not know exactly how

this is my code

```
class RenameModal(ui.Modal, title="Rename"):
  vorname = ui.TextInput(label="Vorname", placeholder="Mein Vorname lautet...", required=True, style=discord.TextStyle.short)
  nachname = ui.TextInput(label="Nachname", placeholder="Mein Nachname lautet...", required=True, style=discord.TextStyle.short)
  spielerid = ui.TextInput(label="ID", placeholder="Meine ID lautet...", required=True, style=discord.TextStyle.short)

  async def on_submit(self, interaction: discord.Interaction):
    await interaction.user.edit(nick=f'{self.vorname} {self.nachname} | {self.spielerid}')

@client.tree.command(name="rename", description="rename")
async def rename(interaction: discord.Interaction):
  await interaction.response.send_modal(RenameModal())
```

as mentioned before i would like to change the nickname of the user after sending the modal

so far I have only tried that and found nothing about it

```
await interaction.user.edit(nick=f'{self.vorname} {self.nachname} | {self.spielerid}')
```

**Comment by tailoric** [comment]
You haven't really described the problem you encounter with your code.
The code you posted should edit the nickname, unless it is longer than 32 characters because that is the maximum length of a nickname on discord.
If you don't get an error check if logging is [set up correctly](https://discordpy.readthedocs.io/en/latest/logging.html) 

Other things to consider. 
- Your bot needs `manage_nicknames` permission
- This command won't work in DMs but will still be available there if synced globally
- you never respond to the interaction which will cause an error on the discord client, you need to reply with something, consider a message that says "changed username successfully to {new nickname}"

**Reply by k3nnyburger** [reply]
Hey thank you for your reply,

when i write this line it says:
`Cannot access member "edit" for type "User"`

When i execute the command in a normal discord channel it says:
```
Traceback (most recent call last):
  File "C:\Users\lepla\AppData\Local\Programs\Python\Python310\lib\site-packages\discord\ui\modal.py", line 186, in _scheduled_task
    await self.on_submit(interaction)
TypeError: RenameModal.on_submit() missing 1 required positional argument: 'member'
```

The bot has adminestrator permissions

**Reply by tailoric** [reply]
> when i write this line it says:

what line? You are showing an error but not what line causes it 
> `Cannot access member "edit" for type "User"`

Yes exactly what I was talking about, this command won't work in DMs because you will receive a user who can't have nicknames.
Your bot can only edit nicknames of member objects, which should be the case when doing the command inside a guild (I think, not 100% sure for the modal).

Please provide your updated code. The error message tells me the initial code block above is not the same as the one that causes your error currently.
Also provide the full traceback not just the last part.

**Reply by k3nnyburger** [reply]
Okay i think i know what you mean
Do you know how to change the nickname from the user on the discord server via the modal?

**Reply by tailoric** [reply, marked as answer]
You have the pieces already here. 
But you are not giving enough context to help you.
The original code does what you want. It will edit the nickname of a server member when done inside a guild.
To protect against it being used outside of a guild (server) you can use the decorator [`@guild_only`](https://discordpy.readthedocs.io/en/latest/interactions/api.html?highlight=guild_only#discord.app_commands.guild_only).
Discord expects you to give a response to the modal submission. 
Editing a user is not a response in regards to the interaction, (discord doesn't know that you did that edit because of the submit button click in the modal) you need to do something with the `interaction.response`. Something you could do is doing `interaction.response.send_message("Changed your nickname", ephemeral=True)` to show a message only visible to the user who used the slash command.

I also quickly tested the code snippet above in my test bot and another thing I have noticed is that this command doesn't work if the member you are trying to edit is higher in the hierarchy than your bot. For example you can never edit the nickname of the server owner or other admins.

If you still encounter problems please give more context to help you and guide you.
Please show all relevant and **current versions** of your code, please show your **entire traceback** not just the last part.

And actually tell us what exactly is not working and what you are expecting to happen instead of "it just fails".

**Reply by k3nnyburger** [reply]
@tailoric Thank you very much for your answer. It wont work because i am the server owner. Now it works. Have a great day 😄
